### PR TITLE
Fix the utilitycost:variable name list error reported in Issue 8455

### LINKED ISSUE #8455: UtilityCost:Variable type list error and mismatch
Issue overview
--------------
UtilityCost:Variable have listed type "Power" (among "Energy", "Power", "Dimensionless", and "Currency") in the Energy+.idd file; however, in the source code it accepts "Demand" rather than "Power". The I/O references list the option as "Demand" (among "Energy", "Demand", "Currency", and "Dimensionless"). 

This mismatch will cause a JSON schema rejection when inputting "Demand" to this field in an .idf file; and a rejection by the input processing code sections when inputting "Power". In neither way would the program run properly. Severe errors will be reported. 

This should be easy to fix by either changing the "Power" option in the .idd feeder file to "Demand", or by changing the code from "Demand" to "Power"--in the latter option a fix can still be made with some more code/doc changes. 

For the users, a temporary workaround would be manually changing the key "Power" under UtilityCostVariable->A3 to "Demand" in the Energy+.idd file. An additional step is needed to: find the "Power" in the enum list of under "UtilityCost:Variable" the Energy+.schema.epJSON and change it to "Demand" as well. In v9.4, this section is originally like this: 
```
                        "variable_type": {
                            "type": "string",
                            "enum": [
                                "",
                                "Currency",
                                "Dimensionless",
                                "Energy",
                                "Power"
                            ],
                            "default": "Dimensionless"
```
Then then change the "Power" there (on Line 191611 in v9.4) to "Demand". 

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version) All
 - Version of EnergyPlus (if using an intermediate build, include SHA): 9.2-9.4
 - Helpdesk ticket number 15607

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jcyuan2020**
Pull request overview
---------------------
 - Fixes #8455  
 - The UtilityCost:Variable input of ``Demand`` is now working properly as expected, and in the same way as the other input types do---``Energy``, ``Dimensionless``, and ``Currency``. The four variable types now also perfectly match the descriptions in the Input/Output Reference document. 

Note that no transition is necessary, because prior to this fix Demand was rejected as an invalid key choice and Power was rejected by the module input processing code.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by Myoldmopar** [comment]
I don't have any issue with this.  Anything remaining before final testing/merge?

**Comment by mjwitte** [comment]
@Myoldmopar Thanks for the review. Please wait for @JasonGlazer to approve this before merging.

**Review by JasonGlazer — APPROVED** [review]

**Comment by mjwitte** [comment]
@jcyuan2020 Please merge in develop one more time.
@Myoldmopar This is ready for final review/merge.

**Comment by jcyuan2020** [comment]
> @mjwitte  @Myoldmopar It is up to date and ready now.

File: tst/EnergyPlus/unit/EconomicTariff.unit.cc
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/EconomicTariff.unit.cc:821]: Hmm, processing multiple IDF snippets in one unit test.  Seems...susceptible to problems later.  That's equivalent to asking EnergyPlus to process two different IDFs.  I would like to see this simply split into different unit tests to ensure they don't accidentally cause problems later.

File: idd/Energy+.idd.in
Comment (Myoldmopar) [inline idd/Energy+.idd.in:96544]: Simple enough.

**Review by Myoldmopar — CHANGES_REQUESTED** [review]
I think I'd like to see the unit test split up per my comments, but otherwise this is good to go.  If someone feels very strongly that the unit test should stay as one and wants to expend the effort and time convincing me, I could possibly be convinced.  Anyway, one more updated set of CI results with all the recent changes in develop and I suspect this can go right in.

File: tst/EnergyPlus/unit/EconomicTariff.unit.cc
Comment (jcyuan2020) [reply tst/EnergyPlus/unit/EconomicTariff.unit.cc:821]: The unit tests are now divided into individual ones. Thanks!

File: idd/Energy+.idd.in
Comment (jcyuan2020) [reply idd/Energy+.idd.in:96544]: Thanks for the review!

**Comment by Myoldmopar** [comment]
Wondeful, thanks!  Building it locally now.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
Tested everything locally, all good.  Ready to merge this.  Thanks @jcyuan2020

**Pull request merged**
